discretisedfield/plotting/mpl.py

```diff
@@ -0,0 +1,873 @@
+import mpl_toolkits.axes_grid1.inset_locator
+import matplotlib.pyplot as plt
+import numpy as np
+
+import discretisedfield.util as dfu
+import ubermagutil.units as uu
+
+
+class Mpl:
+    """Matplotlib-based plotting convenience methods.
+
+    Parameters
+    ----------
+    data : df.Field
+
+        Field sliced with a plane, e.g. field.plane('x').
+
+    """
+
+    def __init__(self, data):
+        # We never allow plotting fields that are not sliced.
+        if not data.mesh.attributes['isplane']:
+            msg = 'The field must be sliced before it can be plotted.'
+            raise ValueError(msg)
+
+        self.data = data  # TODO: Consider renaming data to field.
+        self.planeaxis = dfu.raxesdict[data.mesh.attributes['planeaxis']]
+
+
+    # TODO: I assume the functionality from plot is being moved to plot_simplified?
```

Comment:
`plot_simplified` should completely replace `plot` (and then be renamed to `plot`).